edb/edgeql/parser/grammar/expressions.py

```diff
@@ -956,6 +956,11 @@ class NamedTupleElementList(ListNonterm, element=NamedTupleElement,
     pass
 
 
+class NonEmptySet(Nonterm):
```

Comment:
-1 on doing it this way. 

An `Unexpected {}` error would be confusing to users.  Instead this should be handled directly in `Set.reduce_LBRACE_OptExprList_RBRACE` with an `if not elements: raise SyntaxError("nice error message, with a suggestion")`